grails-app/assets/javascripts/search.js

```diff
@@ -622,6 +622,142 @@ $(document).ready(function() {
         window.location.href = BC_CONF.serverName + "/occurrences/facets/download" + BC_CONF.facetDownloadQuery + '&facets=' + facetName;
     });
 
+    // when open the user preference dlg
+    $('.DQPrefSettingsLink').click(function() {
+        var prefSettings = $('#DQPrefSettings');
+        var userPref = prefSettings.data('userpref-json');
+        var profiles = prefSettings.data('profiles');
+
+        var userProfileEnabled = false;
+
+        // if not disable all
+        if (!userPref.disableAll) {
+            // if preferred profile set
+            var userProfileSet = userPref.dataProfile != null && userPref.dataProfile.length > 0;
+
+            for (var i = 0; i < profiles.length; i++) {
+                if (profiles[i] === userPref.dataProfile) {
+                    userProfileEnabled = true;
+                    break;
+                }
+            }
+        }
+
+        var profileSelect = $('#prefer_profile');
+
+        if (userPref.disableAll) { // if disable all
+            profileSelect.val('disableall-option');
+        } else if (userProfileEnabled && userPref.dataProfile !== null) { // if a profile selected and enabled
+            profileSelect.val(userPref.dataProfile);
+        } else { // if no profile selected or selected profile disabled, use system default
+            profileSelect.val(prefSettings.data('defaultprofilename'));
+        }
+
+        $('#profile_expand').val(userPref.expand ? 'expanded' : 'collapsed');
+    })
+
+    // when submit the user preference dlg
+    $("#submitPref :input.submit").on("click", function(e) {
+        e.preventDefault();
+        var prefSettings = $('#DQPrefSettings');
+        var userPref = prefSettings.data('userpref-json');
+
+        // check user preferred profile
+        var prefProfile = $('#prefer_profile').val();
+        if (prefProfile === 'disableall-option') {
+            userPref.disableAll = true;
+            userPref.dataProfile = null;
+        } else {
+            userPref.disableAll = false;
+            userPref.dataProfile = prefProfile;
+        }
+
+        // set expand
+        userPref.expand = $('#profile_expand').val() === 'expanded';
+        $.cookie.json = true;
+        // if user logged in
+        if (BC_CONF.userId) {
+            // save the dq profile detail expand/collapse state
+            $.cookie(BC_CONF.expandKey, {expand: userPref.expand});
+            $.ajax({
+                url: BC_CONF.serverName + "/user/" + BC_CONF.prefKey,
+                type: "POST",
+                contentType: 'application/json',
+                data: JSON.stringify(userPref),
+                success: applyUserPreference // reload on success
+            }).always(function() {
```

Comment:
Might need to add a `.fail()` or equivalent here to show user error message if this call fails.